### ISSUE TITLE
CPAN::Reporter 1.2003 fails to report about itself on Strawberry Perl 5.14.2.1 [rt.cpan.org #74818]

### ISSUE THREAD

**Issue by xdg** [cpan-testers/CPAN-Reporter#39]
https://rt.cpan.org/Ticket/Display.html?id=74818

    Hi,
    
    tl;dr
    - CPAN::Reporter 1.2003 test fails
    - after force install cannot send report  (tried twice)
    - cannot even fetch existing reports
    
    See details below:
    
    
    I have installed CPAN::Reporter 1.2003 without tests as they were
    failing and then
    I was trying to run "test CPAN::Reporter" and submit the report but that failed.
    
    This is http://strawberryperl.com/package/kmx/p5.14.2.1-RC/strawberry-perl-5.14.2.1-32bit.msi
    
    After upgrading/installing the following packages:
    
    Capture::Tiny                       0.11       => 0.15
    Config::Tiny                                    2.14
    IO::CaptureOutput                               1.1102
    Devel::Autoflush                                0.05
    Test::Reporter                                  1.58
    CPAN::Reporter                                  1.2003     Test
    Failed. "notest install CPAN::Reporer"
    Data::UUID                                      1.217
    Sub::Install                                    0.925
    Data::OptList                                   0.107
    Sub::Exporter                                   0.982
    Data::GUID                                      0.046
    Metabase::Fact::Hash                            0.020
    CPAN::Testers::Report                           1.999001
    Config::Perl::V                                 0.14
    Metabase::Client::Simple                        0.008
    Test::Reporter::Transport::Metabase             1.999008
    Mozilla::CA                         20111025    => 20120118
    Net::HTTP                           6.01        => 6.02
    
    
    
    Currently IPC::Cmd 0.72 is installed. Apparently that caused
    some of the failures.
    
    
    cpan> test CPAN::Reporter
    Database was generated on Thu, 09 Feb 2012 09:17:57 GMT
    Running test for module 'CPAN::Reporter'
    Running make for D/DA/DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
    Checksum for C:\strawberry\cpan\sources\authors\id\D\DA\DAGOLDEN\CPAN-Reporter-1.2003.tar.gz
    ok
    Scanning cache C:\strawberry\cpan\build for sizes
    ............................................................................DONE
    
      CPAN.pm: Building D/DA/DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
    
    Checking if your kit is complete...
    Looks good
    Writing Makefile for CPAN::Reporter
    Writing MYMETA.yml and MYMETA.json
    (C:\strawberry\perl\bin\perl.exe Makefile.PL exited with 0)
    CPAN::Reporter: Makefile.PL result is 'pass', No errors.
    cp lib/CPAN/Reporter.pm blib\lib\CPAN\Reporter.pm
    cp lib/CPAN/Reporter/History.pm blib\lib\CPAN\Reporter\History.pm
    cp lib/CPAN/Reporter/Config.pm blib\lib\CPAN\Reporter\Config.pm
    cp lib/CPAN/Reporter/API.pm blib\lib\CPAN\Reporter\API.pm
    cp lib/CPAN/Reporter/PrereqCheck.pm blib\lib\CPAN\Reporter\PrereqCheck.pm
    cp lib/CPAN/Reporter/FAQ.pm blib\lib\CPAN\Reporter\FAQ.pm
    (C:\strawberry\c\bin\dmake.EXE exited with 0)
    CPAN::Reporter: dmake result is 'pass', No errors.
      DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
      C:\strawberry\c\bin\dmake.EXE -- OK
    Running make test
    C:\strawberry\perl\bin\perl.exe "-MExtUtils::Command::MM" "-e"
    "test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
    t/00-compile.t ................. ok
    t/01_CPAN_Reporter.t ........... ok
    t/02_config_env_vars.t ......... ok
    
    #   Failed test 'normalize id_file: ~/other.json'
    #   at t/03_config_file.t line 76.
    #          got:
    'C:\DOCUME~1\GABORS~1\LOCALS~1\Temp\CPAN-Reporter-testhome-t36Ulol_\.cpanreporter\~\other.json'
    #     expected: '~/other.json'
    # Looks like you failed 1 test of 62.
    t/03_config_file.t .............
    Dubious, test returned 1 (wstat 256, 0x100)
    Failed 1/62 subtests
            (less 2 skipped subtests: 59 okay)
    t/04_option_parsing.t .......... ok
    t/05_prompting.t ............... ok
    t/06_prompt_text.t ............. ok
    t/10_prereq_pm.t ............... ok
    t/11_env_config.t .............. ok
    t/12_toolchain_versions.t ...... ok
    t/13_record_command.t .......... ok
    t/14_command_timeout.t ......... skipped: $ENV{RELEASE_TESTING}
    required for Win32 timeout testing
    t/15_option_validation.t ....... ok
    t/20_report_output.t ........... ok
    t/30_PL_report.t ............... ok
    t/31_PL_report_discard.t ....... ok
    t/40_make_report.t ............. ok
    t/41_make_report_discard.t ..... ok
    t/50_test_report_pass.t ........ ok
    t/51_test_report_fail.t ........ ok
    t/52_test_report_unknown.t ..... ok
    t/53_test_report_na.t .......... ok
    t/54_test_report_split.t ....... ok
    t/55_test_report_discard.t ..... ok
    t/56_test_report_by_harness.t .. ok
    t/57_hang_interrupt.t .......... skipped: $ENV{RELEASE_TESTING}
    required for Win32 timeout testing
    t/60_discard_triggers.t ........ ok
    t/61_bad_dist_names.t .......... ok
    t/62_duplicate_reports.t ....... ok
    t/63_config_send_report.t ...... ok
    t/64_transport.t ............... ok
    t/65_skipfile.t ................ ok
    t/66_have_tested.t ............. ok
    t/67_distfile.t ................ ok
    t/68_is_make.t ................. ok
    
    #   Failed test 'require CPAN::Reporter::Config;'
    #   at t/70_darwin_move_config.t line 60.
    #     Tried to require 'CPAN::Reporter::Config'.
    #     Error:  Bareword "WNOHANG" not allowed while "strict subs" in
    use at C:/strawberry/perl/lib/IPC/Cmd.pm line 417.
    # BEGIN not safe after errors--compilation aborted at
    C:/strawberry/perl/lib/IPC/Cmd.pm line 1285.
    # Compilation failed in require at
    C:\strawberry\cpan\build\CPAN-Reporter-1.2003-J7htKh\blib\lib/CPAN/Reporter/Config.pm
    line 10.
    # BEGIN failed--compilation aborted at
    C:\strawberry\cpan\build\CPAN-Reporter-1.2003-J7htKh\blib\lib/CPAN/Reporter/Config.pm
    line 10.
    # Compilation failed in require at (eval 22) line 2.
    
    #   Failed test 'require CPAN::Reporter::Config;'
    #   at t/70_darwin_move_config.t line 76.
    #     Tried to require 'CPAN::Reporter::Config'.
    #     Error:  Attempt to reload IPC/Cmd.pm aborted.
    # Compilation failed in require at
    C:\strawberry\cpan\build\CPAN-Reporter-1.2003-J7htKh\blib\lib/CPAN/Reporter/Config.pm
    line 10.
    # BEGIN failed--compilation aborted at
    C:\strawberry\cpan\build\CPAN-Reporter-1.2003-J7htKh\blib\lib/CPAN/Reporter/Config.pm
    line 10.
    # Compilation failed in require at (eval 24) line 2.
    
    #   Failed test 'CPAN::Reporter::Config reloaded'
    #   at t/70_darwin_move_config.t line 78.
    
    #   Failed test 'darwin logic: old config-dir removed'
    #   at t/70_darwin_move_config.t line 81.
    
    #   Failed test 'darwin logic: new config-dir created'
    #   at t/70_darwin_move_config.t line 84.
    No such file or directory at t/70_darwin_move_config.t line 87.
    # Looks like you planned 9 tests but ran 7.
    # Looks like you failed 5 tests of 7 run.
    # Looks like your test exited with 2 just after 7.
    t/70_darwin_move_config.t ......
    Dubious, test returned 2 (wstat 512, 0x200)
    Failed 7/9 subtests
    t/71_missing_config.t .......... ok
    t/72_rename_history.t .......... ok
    t/73_autoflush.t ............... ok
    
    Test Summary Report
    -------------------
    t/03_config_file.t           (Wstat: 256 Tests: 62 Failed: 1)
      Failed test:  10
      Non-zero exit status: 1
    t/70_darwin_move_config.t    (Wstat: 512 Tests: 7 Failed: 5)
      Failed tests:  1, 4-7
      Non-zero exit status: 2
      Parse errors: Bad plan.  You planned 9 tests but ran 7.
    Files=39, Tests=1830, 1646 wallclock secs ( 0.43 usr +  0.93 sys =  1.36 CPU)
    Result: FAIL
    Failed 2/39 test programs. 6/1830 subtests failed.
    dmake.EXE:  Error code 255, while making 'test_dynamic'
    (C:\strawberry\c\bin\dmake.EXE test exited with 65280)
    CPAN::Reporter: Test result is 'fail', One or more tests failed.
    CPAN::Reporter: preparing a CPAN Testers report for CPAN-Reporter-1.2003
    Do you want to review or edit the test report? (yes/no) [no]
    Do you want to send the report? (yes/no) [yes]
    CPAN::Reporter: sending test report with 'fail' via Metabase
    CPAN::Reporter: Test::Reporter: error from
    'Test::Reporter::Transport::Metabase:'
    fact submission failed: Status read failed: An invalid argument was
    supplied. at C:\strawberry\perl\site\lib/Metabase/Client/Simple.pm
    line 111
            Metabase::Client::Simple::submit_fact('Metabase::Client::Simple=HASH(0x3421a8c)',
    'CPAN::Testers::Report=HASH(0x370a9b4)') called at
    C:\strawberry\perl\site\lib/Test/Repo
    rter/Transport/Metabase.pm line 132
            Test::Reporter::Transport::Metabase::send('Test::Reporter::Transport::Metabase=HASH(0x1af2e6c)',
    'Test::Reporter=HASH(0x1d309bc)') called at
    C:\strawberry\perl\site\lib/T
    est/Reporter.pm line 271
            eval {...} called at
    C:\strawberry\perl\site\lib/Test/Reporter.pm line 271
            Test::Reporter::send('Test::Reporter=HASH(0x1d309bc)') called
    at C:\strawberry\perl\site\lib/CPAN/Reporter.pm line 492
            CPAN::Reporter::_dispatch_report('HASH(0x2e377a4)') called at
    C:\strawberry\perl\site\lib/CPAN/Reporter.pm line 104
            CPAN::Reporter::grade_test('CPAN::Distribution=HASH(0x181fc3c)',
    'C:\strawberry\c\bin\dmake.EXE test', 'ARRAY(0x27a7f44)', 65280)
    called at C:\strawberry\perl\site\lib/CP
    AN/Reporter.pm line 199
            CPAN::Reporter::test('CPAN::Distribution=HASH(0x181fc3c)',
    'C:\strawberry\c\bin\dmake.EXE test') called at
    C:\strawberry\perl\lib/CPAN/Distribution.pm line 3247
            CPAN::Distribution::test('CPAN::Distribution=HASH(0x181fc3c)')
    called at C:\strawberry\perl\lib/CPAN/Module.pm line 479
            eval {...} called at C:\strawberry\perl\lib/CPAN/Module.pm line 478
            CPAN::Module::rematein('CPAN::Module=HASH(0x181fefc)', 'test')
    called at C:\strawberry\perl\lib/CPAN/Module.pm line 508
            CPAN::Module::test('CPAN::Module=HASH(0x181fefc)') called at
    C:\strawberry\perl\lib/CPAN/Shell.pm line 1797
            CPAN::Shell::rematein('CPAN::Shell', 'test', 'CPAN::Reporter')
    called at C:\strawberry\perl\lib/CPAN/Shell.pm line 1977
            CPAN::Shell::__ANON__('CPAN::Shell', 'CPAN::Reporter') called
    at C:/strawberry/perl/lib/CPAN.pm line 376
            eval {...} called at C:/strawberry/perl/lib/CPAN.pm line 373
            CPAN::shell() called at C:/strawberry/perl/lib/App/Cpan.pm line 295
            App::Cpan::_process_options('App::Cpan') called at
    C:/strawberry/perl/lib/App/Cpan.pm line 364
            App::Cpan::run('App::Cpan') called at C:\strawberry\perl\bin/cpan line 8
    
    
      DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
      C:\strawberry\c\bin\dmake.EXE test -- NOT OK
    //hint// to see the cpan-testers results for installing this module, try:
      reports DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
    Failed during this command:
     DAGOLDEN/CPAN-Reporter-1.2003.tar.gz         : make_test NO
    
    cpan>
    
    
    ==========================================================
    After upgrading to IPC::Cmd 0.76 I tried it again. Now I got:
    
    
    cpan> test CPAN::Reporter
    Running test for module 'CPAN::Reporter'
    Running make for D/DA/DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
    Checksum for C:\strawberry\cpan\sources\authors\id\D\DA\DAGOLDEN\CPAN-Reporter-1.2003.tar.gz
    ok
    Scanning cache C:\strawberry\cpan\build for sizes
    ............................................................................DONE
    
      CPAN.pm: Building D/DA/DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
    
    Checking if your kit is complete...
    Looks good
    Writing Makefile for CPAN::Reporter
    Writing MYMETA.yml and MYMETA.json
    (C:\strawberry\perl\bin\perl.exe Makefile.PL exited with 0)
    CPAN::Reporter: Makefile.PL result is 'pass', No errors.
    cp lib/CPAN/Reporter.pm blib\lib\CPAN\Reporter.pm
    cp lib/CPAN/Reporter/History.pm blib\lib\CPAN\Reporter\History.pm
    cp lib/CPAN/Reporter/Config.pm blib\lib\CPAN\Reporter\Config.pm
    cp lib/CPAN/Reporter/API.pm blib\lib\CPAN\Reporter\API.pm
    cp lib/CPAN/Reporter/PrereqCheck.pm blib\lib\CPAN\Reporter\PrereqCheck.pm
    cp lib/CPAN/Reporter/FAQ.pm blib\lib\CPAN\Reporter\FAQ.pm
    (C:\strawberry\c\bin\dmake.EXE exited with 0)
    CPAN::Reporter: dmake result is 'pass', No errors.
      DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
      C:\strawberry\c\bin\dmake.EXE -- OK
    Running make test
    C:\strawberry\perl\bin\perl.exe "-MExtUtils::Command::MM" "-e"
    "test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
    t/00-compile.t ................. ok
    t/01_CPAN_Reporter.t ........... ok
    t/02_config_env_vars.t ......... ok
    
    #   Failed test 'normalize id_file: ~/other.json'
    #   at t/03_config_file.t line 76.
    #          got:
    'C:\DOCUME~1\GABORS~1\LOCALS~1\Temp\CPAN-Reporter-testhome-cSAMv3P8\.cpanreporter\~\other.json'
    #     expected: '~/other.json'
    # Looks like you failed 1 test of 62.
    t/03_config_file.t .............
    Dubious, test returned 1 (wstat 256, 0x100)
    Failed 1/62 subtests
            (less 2 skipped subtests: 59 okay)
    t/04_option_parsing.t .......... ok
    t/05_prompting.t ............... ok
    t/06_prompt_text.t ............. ok
    t/10_prereq_pm.t ............... ok
    t/11_env_config.t .............. ok
    t/12_toolchain_versions.t ...... ok
    t/13_record_command.t .......... ok
    t/14_command_timeout.t ......... skipped: $ENV{RELEASE_TESTING}
    required for Win32 timeout testing
    t/15_option_validation.t ....... ok
    t/20_report_output.t ........... ok
    t/30_PL_report.t ............... ok
    t/31_PL_report_discard.t ....... ok
    t/40_make_report.t ............. ok
    t/41_make_report_discard.t ..... ok
    t/50_test_report_pass.t ........ ok
    t/51_test_report_fail.t ........ ok
    t/52_test_report_unknown.t ..... ok
    t/53_test_report_na.t .......... ok
    t/54_test_report_split.t ....... ok
    t/55_test_report_discard.t ..... ok
    t/56_test_report_by_harness.t .. ok
    t/57_hang_interrupt.t .......... skipped: $ENV{RELEASE_TESTING}
    required for Win32 timeout testing
    t/60_discard_triggers.t ........ ok
    t/61_bad_dist_names.t .......... ok
    t/62_duplicate_reports.t ....... ok
    t/63_config_send_report.t ...... ok
    t/64_transport.t ............... ok
    t/65_skipfile.t ................ ok
    t/66_have_tested.t ............. ok
    t/67_distfile.t ................ ok
    t/68_is_make.t ................. ok
    t/70_darwin_move_config.t ...... ok
    t/71_missing_config.t .......... ok
    t/72_rename_history.t .......... ok
    t/73_autoflush.t ............... ok
    
    Test Summary Report
    -------------------
    t/03_config_file.t           (Wstat: 256 Tests: 62 Failed: 1)
      Failed test:  10
      Non-zero exit status: 1
    Files=39, Tests=1832, 1756 wallclock secs ( 0.42 usr +  1.09 sys =  1.51 CPU)
    Result: FAIL
    Failed 1/39 test programs. 1/1832 subtests failed.
    dmake.EXE:  Error code 255, while making 'test_dynamic'
    (C:\strawberry\c\bin\dmake.EXE test exited with 65280)
    CPAN::Reporter: Test result is 'fail', One or more tests failed.
    CPAN::Reporter: preparing a CPAN Testers report for CPAN-Reporter-1.2003
    Do you want to review or edit the test report? (yes/no) [no]
    Do you want to send the report? (yes/no) [yes]
    CPAN::Reporter: sending test report with 'fail' via Metabase
    CPAN::Reporter: Test::Reporter: error from
    'Test::Reporter::Transport::Metabase:'
    fact submission failed: Status read failed: An invalid argument was
    supplied. at C:\strawberry\perl\site\lib/Metabase/Client/Simple.pm
    line 111
            Metabase::Client::Simple::submit_fact('Metabase::Client::Simple=HASH(0x342a02c)',
    'CPAN::Testers::Report=HASH(0x370c514)') called at
    C:\strawberry\perl\site\lib/Test/Repo
    rter/Transport/Metabase.pm line 132
            Test::Reporter::Transport::Metabase::send('Test::Reporter::Transport::Metabase=HASH(0x1af35e4)',
    'Test::Reporter=HASH(0x1d31724)') called at
    C:\strawberry\perl\site\lib/T
    est/Reporter.pm line 271
            eval {...} called at
    C:\strawberry\perl\site\lib/Test/Reporter.pm line 271
            Test::Reporter::send('Test::Reporter=HASH(0x1d31724)') called
    at C:\strawberry\perl\site\lib/CPAN/Reporter.pm line 492
            CPAN::Reporter::_dispatch_report('HASH(0x2e68ae4)') called at
    C:\strawberry\perl\site\lib/CPAN/Reporter.pm line 104
            CPAN::Reporter::grade_test('CPAN::Distribution=HASH(0x182000c)',
    'C:\strawberry\c\bin\dmake.EXE test', 'ARRAY(0x27b69b4)', 65280)
    called at C:\strawberry\perl\site\lib/CP
    AN/Reporter.pm line 199
            CPAN::Reporter::test('CPAN::Distribution=HASH(0x182000c)',
    'C:\strawberry\c\bin\dmake.EXE test') called at
    C:\strawberry\perl\lib/CPAN/Distribution.pm line 3247
            CPAN::Distribution::test('CPAN::Distribution=HASH(0x182000c)')
    called at C:\strawberry\perl\lib/CPAN/Module.pm line 479
            eval {...} called at C:\strawberry\perl\lib/CPAN/Module.pm line 478
            CPAN::Module::rematein('CPAN::Module=HASH(0x18202cc)', 'test')
    called at C:\strawberry\perl\lib/CPAN/Module.pm line 508
            CPAN::Module::test('CPAN::Module=HASH(0x18202cc)') called at
    C:\strawberry\perl\lib/CPAN/Shell.pm line 1797
            CPAN::Shell::rematein('CPAN::Shell', 'test', 'CPAN::Reporter')
    called at C:\strawberry\perl\lib/CPAN/Shell.pm line 1977
            CPAN::Shell::__ANON__('CPAN::Shell', 'CPAN::Reporter') called
    at C:/strawberry/perl/lib/CPAN.pm line 376
            eval {...} called at C:/strawberry/perl/lib/CPAN.pm line 373
            CPAN::shell() called at C:/strawberry/perl/lib/App/Cpan.pm line 295
            App::Cpan::_process_options('App::Cpan') called at
    C:/strawberry/perl/lib/App/Cpan.pm line 364
            App::Cpan::run('App::Cpan') called at C:\strawberry\perl\bin/cpan line 8
    
    
    
    ================================
    
    
    cpan> reports DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
    Distribution: D/DA/DAGOLDEN/CPAN-Reporter-1.2003.tar.gz
    Fetching 'http://www.cpantesters.org/show/CPAN-Reporter.yaml'...DONE
    
    Catching error: "CPAN::Exception::yaml_process_error=HASH(0x3805fec)"
    at C:/strawberry/perl/lib/CPAN.pm line 392
            CPAN::shell() called at C:/strawberry/perl/lib/App/Cpan.pm line 295
            App::Cpan::_process_options('App::Cpan') called at
    C:/strawberry/perl/lib/App/Cpan.pm line 364
            App::Cpan::run('App::Cpan') called at C:\strawberry\perl\bin/cpan line 8
    
    
    
    Gabor

**Comment by twata1** [comment]
Hi,

Information that might be useful.

Bug #97166 for IO-Socket-SSL: 1.995 breaks CPAN Reporter with IO::Socket::IP 0.31
https://rt.cpan.org/Public/Bug/Display.html?id=97166



Thank you,

--
twata
